sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -397,7 +398,8 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
     val tableDesc = CatalogTable(
       identifier = table,
       tableType = tableType,
-      storage = storage.copy(locationUri = customLocation),
+      storage = storage.copy(locationUri = customLocation.map{ loc =>
```

Comment:
this pattern appears many times in this PR, how about we create a util method `toURI` in `CatalogUtils` and add comments there?